tests/orders/__init__.py

```diff
@@ -77,7 +78,21 @@ def take_profit_buy_order(event_loop, simulated_trader):
     return TakeProfitOrder(trader_instance, side=TradeOrderSide.BUY)
 
 
-@pytest.fixture()
+@pytest.yield_fixture()
 def take_profit_limit_order(event_loop, simulated_trader):
+    _, exchange_manager, trader_instance = simulated_trader
+    yield TakeProfitLimitOrder(trader_instance)
+    exchange_manager.exchange_personal_data.orders_manager.clear()
```

Comment:
why having to clear it before exchange stop ?

Reply:
To kill the order related price_event tasks to prevent CancelledErrors

Reply:
I mean exchange_manager.clear() should already call this, it's not working ?

Reply:
The problem here is not the initial `TakeProfitLimitOrder` clear(). But it's artificial order when created that is added in `exchange_manager.exchange_personal_data.orders_manager `orders list.